server/rpc.go

```diff
@@ -467,18 +477,52 @@ func SubmitSegment(bcast Broadcaster, seg *stream.HLSSegment) {
 	req.Header.Set("Content-Type", "video/MP2T")
 
 	glog.Infof("Submitting segment %v : %v bytes", seg.SeqNo, len(seg.Data))
+	start := time.Now()
 	resp, err := hc.Do(req)
+	uploadDur := time.Since(start)
 	if err != nil {
 		glog.Error("Unable to submit segment ", seg.SeqNo, err)
+		if monitor.Enabled {
+			monitor.LogSegmentUploadFailed(nonce, seg.SeqNo, err.Error())
+		}
+		return
+	}
+	if resp.StatusCode != 200 {
```

Comment:
Maybe we could do this check after the response body has been read, since we read the body in both the 200 and non-200 cases?